packages/express/src/spec/4.1-structure/4.1.24-http-method.spec.ts

```diff
@@ -0,0 +1,214 @@
+                    value: factory.primitiveValue({
+                      typeName: factory.primitiveLiteral('string'),
+                    }),
+                  }),
+                ],
+                returns: factory.returnValue({
+                  value: factory.primitiveValue({
+                    typeName: factory.primitiveLiteral('number'),
+                  }),
+                }),
+              }),
+            ],
+            protocols: factory.protocols({
+              http: [
+                factory.httpRoute({
+                  pattern: factory.stringLiteral('/widgets/stream'),
+                  methods: [
+                    factory.httpMethod({
+                      name: factory.stringLiteral('streamWidgets'),
+                      verb: factory.httpVerbLiteral('get'),
+                      responseMediaTypes: [
+                        factory.stringLiteral('text/event-stream'),
+                      ],
+                      parameters: [
+                        factory.httpParameter({
+                          name: factory.stringLiteral('a'),
+                          location: factory.httpLocationLiteral('query'),
+                        }),
+                      ],
+                    }),
+                  ],
+                }),
+              ],
+            }),
+          }),
+        ],
+      });
+
+      // ACT
+      const result = await generator(service);
+
+      const handlers = result.find((file) => file.path.includes('handlers.ts'));
+
+      console.log(handlers?.contents);
+
+      // ASSERT
+      expect(handlers?.contents).toContainAst(`
+        /** GET /widgets/stream */
+        export const handleStreamWidgets =
+          (
+            getService: (req: Request, res: Response) => types.WidgetsService,
+          ): expressTypes.StreamWidgetsRequestHandler =>
+          async (req, res, next) => {
+            // Set response headers for streaming
+            res.setHeader('Content-Type', 'text/event-stream');
+            res.setHeader('Cache-Control', 'no-cache');
+            res.setHeader('Connection', 'keep-alive');
+
+            const closeHandler = () => {
+              res.end();
+            };
+            
+            req.on('close', closeHandler);
+            req.on('finish', closeHandler);
+
+            try {
+              // Parse parameters from request
+              const params: types.StreamWidgetsParams =
+                schemas.StreamWidgetsParamsSchema.parse({
+                  a: req.query.a,
+                });
+        
+              // Execute service method
+              const service = getService(req, res);
+
+              const stream = await service.streamWidgets(params);
+              for await (const event of stream) {
+                res.write(\`data: \${JSON.stringify(event)}\\n\\n\`);
+              }
+              closeHandler();
+            } catch (err) {
+              closeHandler();
```

Comment:
We might want to consider moving `closeHandler()` to the finally block. Not sure it has to precede the call to `next()`.